rest/server_context.go

```diff
@@ -41,6 +41,7 @@ const DefaultStatsLogFrequencySecs = 60
 // This struct is accessed from HTTP handlers running on multiple goroutines, so it needs to
 // be thread-safe.
 type ServerContext struct {
+	usingTLS          bool
```

Comment:
I'm not sure we need to add another field in the ServerContext struct, but I had an issue with the naming too, which could cause quite a bit of confusion. It's not strictly related to TLS, but more "are we going to be issuing secure cookies".

I think it's reasonable to keep the config value within the ServerConfig along with a method that returns the value (mentioned above).

Reply:
Yeah. Perfectly valid. Initially this variable was intended to only mark whether TLS was used but when included with the config it obviously changes its use. 
Will add a method instead of field.